packages/backend-core/src/sql/sql.ts

```diff
@@ -42,27 +42,28 @@ const envLimit = environment.SQL_MAX_ROWS
   : null
 const BASE_LIMIT = envLimit || 5000
 
-function likeKey(client: string | string[], key: string): string {
-  let start: string, end: string
+// Takes a string like foo and returns a quoted string like [foo] for SQL Server
+// and "foo" for Postgres.
+function quote(client: SqlClient, str: string) {
   switch (client) {
-    case SqlClient.MY_SQL:
-      start = end = "`"
-      break
     case SqlClient.SQL_LITE:
     case SqlClient.ORACLE:
     case SqlClient.POSTGRES:
-      start = end = '"'
-      break
+      return `"${str}"`
     case SqlClient.MS_SQL:
-      start = "["
-      end = "]"
-      break
+      return `[${str}]`
     default:
-      throw new Error("Unknown client generating like key")
+      return `\`${str}\``
   }
-  const parts = key.split(".")
-  key = parts.map(part => `${start}${part}${end}`).join(".")
+}
+
+// Takes a string like a.b.c and returns a quoted identifier like [a].[b].[c]
+// for SQL Server and `a`.`b`.`c` for MySQL.
+function quotedIdentifier(client: SqlClient, key: string): string {
```

Comment:
This is just `likeKey` from before renamed to be more general (because I'm using it in a context that has nothing to do with `like` queries). It is logically the same as `likeKey` was, except split into another `quote` function so I can use that logic elsewhere.

Reply:
This function name is not really verbose. Should this not be `parseIdentifier` or something similar? And is this quoted, if it returns square brackets or others?

Reply:
I'm not sure what a better word would be than "quoted". I don't really have strong feelings about the name other than `likeKey` was incorrect after the way I used it in this PR. Happy to go with `parseIdentifier` if you'd like, though I'm not sure that's more clear.

Reply:
I was just a little bit confused, as technically it does not always quotes the id